internal/providers/pluginfw/resources/user/data_users.go

```diff
@@ -0,0 +1,113 @@
+package user
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+func DataSourceUsers() datasource.DataSource {
+	return &UsersDataSource{}
+}
+
+var _ datasource.DataSourceWithConfigure = &UsersDataSource{}
+
+type UsersDataSource struct {
+	Client *common.DatabricksClient
+}
+
+type UsersInfo struct {
+	DisplayNameContains string     `json:"display_name_contains,omitempty" tf:"computed"`
+	UserNameContains    string     `json:"user_name_contains,omitempty" tf:"computed"`
+	Users               []iam.User `json:"users,omitempty" tf:"computed"`
```

Comment:
You probably need to use `iam_tf.User` here instead of `iam.User`, since this structure needs to use the types defined in the `types` module of the plugin framework in order to work with the `Get()` and `Set()` methods of the Plan.

Reply:
Done 👍